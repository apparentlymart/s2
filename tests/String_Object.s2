#-*-c++-*-

layerinfo "type" = "core";
layerinfo "majorversion" = 0;

class string { 
    function repeat () : string;
}

function string::repeat () : string
{
    return $this + $this;
}

function main ()
{
    var string b = "foo ";
    var string r = "";
    $r = $b->repeat();
    $r = $r->repeat();

    println "repeated: $r";
}

