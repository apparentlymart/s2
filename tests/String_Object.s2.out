repeated: foo foo foo foo 
